sources/api/shibaken/src/main.rs

```diff
@@ -60,17 +60,23 @@ fn fetch_imds_session_token(client: &Client) -> Result<String> {
     Ok(imds_session_token)
 }
 
-/// Helper to fetch data from IMDS. Taken from pluto.
-fn get_text_from_imds(client: &Client, uri: &str, session_token: &str) -> Result<String> {
-    client
+/// Helper to fetch data from IMDS. Inspired by pluto.
+fn fetch_from_imds(client: &Client, uri: &str, session_token: &str) -> Result<Option<String>> {
+    let response = client
         .get(uri)
         .header("X-aws-ec2-metadata-token", session_token)
         .send()
-        .context(error::ImdsRequest { method: "GET", uri })?
-        .error_for_status()
-        .context(error::ImdsResponse { uri })?
-        .text()
-        .context(error::ImdsText { uri })
+        .context(error::ImdsRequest { method: "GET", uri })?;
+    if response.status().as_u16() == 404 {
+        return Ok(None);
+    }
+    Ok(Some(
+        response
+            .error_for_status()
+            .context(error::ImdsResponse { uri })?
+            .text()
+            .context(error::ImdsText { uri })?,
+    ))
```

Comment:
clippy should have suggested this. I guess. 
```suggestion
    if response.status().as_u16() == 404 {
         Ok(None)
    } else {
        Ok(Some(
        response
            .error_for_status()
            .context(error::ImdsResponse { uri })?
            .text()
            .context(error::ImdsText { uri })?
          )))
    }
```

Reply:
It's a common pattern to return early during an error check, and to keep the happy path clear.  It makes it easier and clearer during later changes.